pkg/ctl/install/install.go

```diff
@@ -0,0 +1,435 @@
+		k8sClientSet: k8sClientSet,
+	}
+	return fi, nil
+}
+
+
+func (fi *fluxInstaller) run(ctx context.Context) error {
+	manifests, err := getFluxManifests(fi.opts.templateParams, fi.k8sClientSet)
+	if err != nil {
+		return fmt.Errorf("failed to create flux manifests: %s", err)
+	}
+
+
+	logger.Info("Cloning %s", fi.opts.templateParams.GitURL)
+	cloneDir, err := fi.cloneRepo(ctx)
+	if err != nil {
+		return fmt.Errorf("cannot clone repository %s: %s", fi.opts.templateParams.GitURL, err)
+	}
+	cleanCloneDir := false
+	defer func() {
+		if cleanCloneDir {
+			os.RemoveAll(cloneDir)
+		} else {
+			logger.Critical("You may find the local clone of %s used by eksctl at %s",
+				fi.opts.templateParams.GitURL,
+				cloneDir)
+		}
+	}()
+
+	logger.Info("Writing Flux manifests")
+	fluxManifestDir := filepath.Join(cloneDir, fi.opts.gitFluxPath)
+	if err := writeFluxManifests(fluxManifestDir, manifests); err != nil {
+		return err
+	}
+
+	if fi.opts.amend {
+		logger.Info("Stopping to amend the the Flux manifests, please exit the shell when done.")
+		if err := runShell(fluxManifestDir); err != nil {
+			return err
+		}
+	}
+
+	// If we need to create a Namespace, we do it synchronously before invoking kubectl apply.
+	// Otherwise there is a race between the creation of the namespace and the other resources living in it
+	if _, ok := manifests[namespaceFileName]; ok {
+		logger.Info("Creating namespace %s", fi.opts.templateParams.Namespace)
+		if err:= createNamespaceSynchronously(fi.k8sClientSet, fi.opts.templateParams.Namespace); err != nil {
+			return err
+		}
+	}
+
+	logger.Info("Installing Flux into the cluster")
+	if err := fi.applyManifestsInDir(ctx, fluxManifestDir); err != nil {
+		return err
+	}
+
+	logger.Info("Waiting for Flux to start")
+	fluxSSHKey, err :=  waitForFluxToStart(ctx, fi.opts, fi.k8sRestConfig, fi.k8sClientSet)
+	if err != nil {
+		return err
+	}
+	logger.Info("Flux started successfully")
+
+	logger.Info("Committing and pushing Flux manifests to %s", fi.opts.templateParams.GitURL)
+	if err := fi.addCommitAndPushFluxManifests(ctx, cloneDir); err != nil {
+		return err
+	}
+	cleanCloneDir = true
+
+	logger.Info("In order for Flux to operate properly make sure it has SSH access to %s",
+		fi.opts.templateParams.GitURL)
+	logger.Info("Flux's current Public SSH key is:\n%s", fluxSSHKey.Key)
+
+	return nil
+}
+
+func (fi *fluxInstaller) cloneRepo(ctx context.Context) (string, error){
+	cloneDir, err := ioutil.TempDir(os.TempDir(), "eksctl-install-flux-clone")
+	if err != nil {
+		return "", fmt.Errorf("cannot create temporary directory: %s", err)
+	}
+	cloneCtx, cloneCtxCancel := context.WithTimeout(ctx, fi.opts.timeout)
+	defer cloneCtxCancel()
+	args := []string{"clone", "-b", fi.opts.templateParams.GitBranch, fi.opts.templateParams.GitURL, cloneDir}
+	err = runGitCmd(cloneCtx,cloneDir, args...)
+	return cloneDir, err
+}
+
+func (fi *fluxInstaller) addCommitAndPushFluxManifests(ctx context.Context, cloneDir string) error {
+	// Add
```

Comment:
I see why this comment is here, but I find it more confusing then helpful, function name is already pretty good.

Reply:
Changed to `gid add` for, hopefully, more clarity. I still find these helpful, given smaller methods for the individual operations wouldn't make much sense because of Go's verbosity.